templates/stamp/stamp.json

```diff
@@ -380,7 +380,7 @@
 
         "installerUtilitiesScript": "utilities.sh",
         "mongoDbInstallerScript": "[concat('mongodb-', variables('supportedOSName'), '-install.sh')]",
-        "mongoDbInstallerCommand": "[concat('bash ', variables('mongoDbInstallerScript'), ' -i ', variables('mongoMachineSettings').installerBaseUrl, ' -b ', variables('mongoMachineSettings').installerPackages, ' -r ', variables('mongoReplicaSetName'), ' -k ', variables('mongoReplicaSetKey'), ' -u ', parameters('mongoServerAdminUserName'), ' -p ', parameters('mongoServerAdminPassword'), ' -x ', variables('networkSettings').serverIpPrefix, ' -n ', variables('networkSettings').mongoDataNodeCount)]",
+        "mongoDbInstallerCommand": "[concat('bash ', variables('mongoDbInstallerScript'), ' -i ', variables('mongoMachineSettings').installerBaseUrl, ' -b ', variables('mongoMachineSettings').installerPackages, ' -r ', variables('mongoReplicaSetName'), ' -k ', variables('mongoReplicaSetKey'), ' -u ', parameters('mongoServerAdminUserName'), ' -p \"', parameters('mongoServerAdminPassword'), '\" -x ', variables('networkSettings').serverIpPrefix, ' -n ', variables('networkSettings').mongoDataNodeCount)]",
```

Comment:
there unfortunately isn't a good way of using single quotations here (due to restrictions in json and concat). a workaround is to base64() encode and decode it within the bash scripts. this is similar to how this problem was solved elsewhere https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/issues/192
there unfortunately isn't a good way of using single quotations here (due to restrictions in json and concat). a workaround is to base64() encode here and then decode it within the bash scripts. this is similar to how this problem was solved elsewhere https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/issues/192 . i'll give that a shot